presto-hive/src/test/java/com/facebook/presto/hive/TestHiveClustering.java

```diff
@@ -88,4 +96,49 @@ public void testCreateTable()
         MaterializedResult result = computeActual(query);
         assertEquals(getOnlyElement(result.getOnlyColumnAsSet()), true);
     }
+
+    @Test
+    public void testCreateTableAsSelect()
+    {
+        String query = (
+                "CREATE TABLE hive_bucketed.tpch_bucketed.customer_copied \n" +
+                        " WITH (" +
+                        "    format = 'TEXTFILE',\n" +
+                        "    clustered_by = array['custkey'], \n" +
+                        "    cluster_count = array[11], \n" +
+                        "    distribution = array['150', '300', '450', '600', '750', '900', '1050', '1200', '1350']) \n" +
+                        " AS SELECT * FROM customer");
+
+        MaterializedResult result = computeActual(query);
+        assertEquals(result.getOnlyValue(), 1500L);
+
+        query = "SELECT COUNT(*) FROM hive_bucketed.tpch_bucketed.customer_copied";
+        result = computeActual(query);
+        assertEquals(result.getOnlyValue(), 1500L);
+
+        query = "SELECT custkey, COUNT(*) FROM hive_bucketed.tpch_bucketed.customer_copied GROUP BY custkey ORDER BY custkey LIMIT 1 ";
+        result = computeActual(query);
+        assertEquals(result.getRowCount(), 1L);
+        assertEquals(result.getMaterializedRows().get(0).getField(1), 1L);
+    }
+
+    @Test
+    public void testInsert()
+    {
+        String query = (
+                "Insert INTO hive_bucketed.tpch_bucketed.customer_copied \n" +
+                "SELECT * from customer");
+
+        MaterializedResult result = computeActual(query);
+        assertEquals(result.getOnlyValue(), 1500L);
+
+        query = "SELECT COUNT(*) FROM hive_bucketed.tpch_bucketed.customer_copied";
+        result = computeActual(query);
+        assertEquals(result.getOnlyValue(), 3000L);
+
+        query = "SELECT custkey, COUNT(*) FROM hive_bucketed.tpch_bucketed.customer_copied GROUP BY custkey ORDER BY custkey LIMIT 1 ";
```

Comment:
An interesting test would be creating two tables: one with clustering and one without. Then use the same select query with some proper filters against these two tables. Then we should expect the one against clustering table will read fewer rows and data.

Reply:
How to check the query stats to get the rows and data read?